tests/test_options_parsers.py

```diff
@@ -0,0 +1,397 @@
+        return contact + mechanics_amg
+    if model_kind == "TH":
+        return interface_flow + thermal_flow
+    if model_kind == "HM":
+        return contact + fixed_stress + isothermal_flow + mechanics_amg
+    if model_kind == "THM":
+        return contact + interface_flow + fixed_stress + thermal_flow + mechanics_amg
+
+    raise NotImplementedError
+
+
+def find_in_petsc_output(pattern: str, petsc_stdout: str):
+    # \s* at the start of each new line means arbitrary number of whitespaces.
+    pattern = "\n".join(r"\s*" + line for line in pattern.splitlines())
+    return re.findall(pattern, petsc_stdout)
+
+
+def test_ksp_none(petsc_stdout: str, model_kind: str):
+    """Ensure that the default preconditioners do not involve inner Krylov solvers.
+
+    In PETSc, this is done by replacing the Krylov solver with "preonly". We also count
+    "preonly" to estimate the number of inner preconditioners.
+
+    """
+    matches = find_in_petsc_output(r"KSP Object:.*\n *type: (.+)", petsc_stdout)
+    if model_kind in ["flow", "mechanics"]:
+        assert len(matches) == 2
+    elif model_kind == "HM":
+        assert len(matches) == 6
+    elif model_kind == "TH":
+        assert len(matches) == 4
+    elif model_kind == "THM":
+        assert len(matches) == 8
+    else:
+        raise NotImplementedError(model_kind)
+
+    assert all(x == "preonly" for x in matches)
+
+
+def test_petsc_options(petsc_stdout: str, patterns_to_compare):
+    """Compare the output of `pc.view()` - PETSc report of how the preconditioner is
+    configured, to the expected values. We search for a few key lines in the report.
+
+    """
+    for pattern in patterns_to_compare:
+        found = find_in_petsc_output(pattern, petsc_stdout)
+        assert len(found) >= 1, f"Not found:\n{pattern}\n\n{petsc_stdout}\n\n"
+
+
+def test_pass_user_options(
+    capfd,  # This is a pytest object to capture the os-level stdout, needed for PETSc.
+    jacobian: BlockMatrixStorage,
+    pc: PETSc.PC,
+    model: pp.PorePyModel,
+    dof_manager: DofManager,
+    solvers: list[SinglePhysicsPreconditioner],
+    model_kind: str,
+) -> str:
+    if model_kind == "mechanics":
+        expected_group_name = "mechanics"
+    else:
+        expected_group_name = "mass_balance"
+
+    user_options = {
+        expected_group_name: {
+            "pc_type": "jacobi",
+        }
+    }
+
+    preconditioner_scheme = MultiPhysicsPreconditioner(
+        components=solvers, dof_manager=dof_manager, model=model
+    )
+    preconditioner_scheme.configure(
+        bmat=jacobian,
+        pc=pc,
+        user_options=user_options,
+        precond_list=solvers,
+    )
+
+    pc.view()
+    petsc_stdout, _ = capfd.readouterr()
+
+    pattern = "PC Object: .* 1 MPI process\ntype: jacobi"
+
+    matches = find_in_petsc_output(pattern, petsc_stdout)
+    assert len(matches) == 1, f"Not found:\n{pattern}\n\n{petsc_stdout}\n\n"
+
+
+def test_petsc_ksp_scheme(
```

Comment:
I get errors in total when running this locally. My first guess is that our petsc versions do not match. I have 3.22.3. What do you think?

Reply:
```
>>> petsc4py.__version__
'3.20.2'
```

Reply:
Let's leave this as it is for now. When we get GH actions up and running, there will be a single truth which may change upon updates of the PETSc version, but that we have to live with. 

Feel free to resolve this, and the related below comments, after reading.